lib/tasks/BackbeatTask.js

```diff
@@ -1,35 +1,16 @@
 const BackOff = require('backo');
-const joi = require('joi');
-
-const BACKOFF_PARAMS = { min: 1000, max: 300000, jitter: 0.1, factor: 1.5 };
-
-const backbeatTaskConfigJoi = {
-    retryTimeoutS: joi.number().default(300),
-};
 
 /**
  * @class BackbeatTask
  */
 class BackbeatTask {
-    /**
-     * @constructor
-     * @param {Object} [config] - config object containing various
-     *   task-specific configuration
-     * @param {Number} [config.retryTimeoutS] - timeout of retries, in
-     *   seconds
-     */
-    constructor(config) {
-        this.config = joi.attempt(config || {}, backbeatTaskConfigJoi,
-                                  'invalid backbeat task config');
-    }
 
     retry(args, done) {
         const { actionDesc, logFields,
                 actionFunc, shouldRetryFunc, onRetryFunc, log } = args;
-        const backoffCtx = new BackOff(BACKOFF_PARAMS);
+        const backoffCtx = new BackOff(this.retryParams.backoff);
```

Comment:
Since we don't provide the retry params in the constructor anymore, I would rather populate them with default values in the constructor, and provide a setter to change them. That would avoid taking the risk of forgetting to set the retry params and raising an exception.